ONSdigital/dp-frontend-search-controller
# Feature/implement date fieldset

### What

- Converted date fields to use the [dp renderer date fieldset](https://github.com/ONSdigital/dp-renderer/blob/main/PATTERNS.md#date-fieldset)
- Removed redundant copy and code 
- Added new copy
- Added unit tests to cover existing mapping logic
- Updated go version
- Update go modules

### How to review

Sense check
Tests pass
Media review
_Optionally_
  - Pull this branch and run app `make debug ENABLE_REWORKED_DATA_AGGREGATION_PAGES=true`
  - Run this branch and this [branch](https://github.com/ONSdigital/dp-design-system/tree/feature/2036-implement-date-fieldset) of `dp-design-system` (in PR https://github.com/ONSdigital/dp-design-system/pull/207)
  - Port forward the api router to sandbox `dp ssh sandbox web 1 -p 23200:localhost:10800`
  - Go to the following routes:
     - http://localhost:25000/alladhocs
     - http://localhost:25000/datalist
     - http://localhost:25000/publishedrequests
     - http://localhost:25000/staticlist
     - http://localhost:25000/topicspecificmethodology
     - http://localhost:25000/timeseriestool
     - http://localhost:25000/publications
     - http://localhost:25000/allmethodologies
   - Check the date fields still look good

#### Media

https://github.com/ONSdigital/dp-frontend-search-controller/assets/19624419/8aa88adb-89ce-4dd9-94b3-7569ed426edb

### Who can review

Frontend go dev
